### PR TITLE
Fix CMake build of TDNF

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This pull request addresses the issues I discovered while updating packaging for TDNF to the latest revision in the dev branch.

The major changes are:

* CMake is now using `GNUInstallDirs` so that the correct target directories are used on various platforms
* Header files are now being installed so that other projects can link to TDNF
* Project versioning is now set through CMake so there's a proper fallback when Git data isn't available
* The pkgconfig files have been restored and are configured and installed by CMake
* Fix includes in Python binding sources so that it builds on a system that doesn't already have TDNF installed in the build environment
* CI has been reworked to add a Fedora-based build and build the Python module

This should fix all the issues I've discovered so far...

### CODE REVIEW

**Comment by ppadmavilasom** [comment]
@Conan-Kudo thanks! appreciate your help

**Comment by ppadmavilasom** [comment]
@Conan-Kudo if we add an rpm build and an isolated dependent installation based on the built rpm, we can identify the pkg-config issues. do you see other test gaps here?

**Comment by Conan-Kudo** [comment]
@ppadmavilasom I think that should mostly cover it. If we add a build using Fedora in here, for example, I think we can make sure things don't break like this again.

**Comment by ppadmavilasom** [comment]
> @ppadmavilasom I think that should mostly cover it. If we add a build using Fedora in here, for example, I think we can make sure things don't break like this again.

Sure. we will add a build using fedora in addition to the one that uses photon now (as well as adding a dependency test via rpm installs or  client code compile from a newly installed chroot)

**Comment by Conan-Kudo** [comment]
@ppadmavilasom I've added a Fedora CI build of TDNF, though I have not added an RPM spec yet...

File: ci/Dockerfile.fedora [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM fedora:latest
+
+MAINTAINER csiddharth@vmware.com
+
+
+RUN dnf -y upgrade
```
Comment (ppadmavilasom) [inline ci/Dockerfile.fedora]: -q as well please

Comment (Conan-Kudo) [reply ci/Dockerfile.fedora]: Why do quiet? Wouldn't that make it look like the CI is doing nothing for a long period of time?

Comment (Conan-Kudo) [reply ci/Dockerfile.fedora]: Anyway, I added the quiet option.

**Review by ppadmavilasom — APPROVED** [review]
LGTM. Thanks for your help.

**Comment by Conan-Kudo** [comment]
@ppadmavilasom This should be good to merge now. We have significantly more coverage than we did before, which is a major step up. There's certainly other improvements to make, but let's get this merged so that it doesn't keep getting bigger. 😄

**Pull request merged**